Lib/test/datetimetester.py

```diff
@@ -7270,8 +7297,103 @@ def test_update_type_cache(self):
                 assert isinstance(_datetime.timezone.utc, _datetime.tzinfo)
                 del sys.modules['_datetime']
             """)
+        res = script_helper.assert_python_ok('-c', script)
+        self.assertFalse(res.err)
+
+    def test_static_type_at_shutdown1(self):
+        # gh-132413
+        script = textwrap.dedent("""
+            import sys
+            import _datetime
+
+            def gen():
+                try:
+                    yield
+                finally:
+                    # Exceptions are ignored here
+                    assert not sys.modules
+                    td = _datetime.timedelta(days=1)
+                    assert td.days == 1
+                    assert not sys.modules
+
+            it = gen()
+            next(it)
+        """)
+        res = script_helper.assert_python_ok('-c', script)
+        self.assertFalse(res.err)
+
+    def test_static_type_at_shutdown2(self):
+        script = textwrap.dedent("""
+            import sys
+            from _datetime import timedelta
+
+            def gen():
+                try:
+                    yield
+                finally:
+                    assert not sys.modules
+                    td = timedelta(days=1)
+                    assert td.days == 1
+                    assert not sys.modules
+
+            it = gen()
+            next(it)
+        """)
+        res = script_helper.assert_python_ok('-c', script)
+        self.assertFalse(res.err)
+
+    def test_static_type_at_shutdown3(self):
+        script = textwrap.dedent("""
+            timedelta = _testcapi.get_capi_types()['timedelta']
+
+            def gen():
+                try:
+                    yield
+                finally:
+                    timedelta(days=1)
+
+            it = gen()
+            next(it)
+        """)
+        res = CapiTest.assert_python_ok_in_subinterp(self, script, setup='')
+        self.assertIn(b'ImportError: sys.meta_path is None', res.err)
```

Comment:
This is possible after switching or restarting the interpreter, so the PR seems not to be advantageous for now.

Reply:
Fixed by c7fc55e5b1ec73c2fc26ffe0cc9a606fd1599916.